rcl_yaml_param_parser/src/parser.c

```diff
@@ -824,71 +814,77 @@ static void * get_value(
 ///
 /// Parse the value part of the <key:value> pair
 ///
-static rcl_ret_t parse_value(
+static rcutils_ret_t parse_value(
   const yaml_event_t event,
   const bool is_seq,
   data_types_t * seq_data_type,
   rcl_params_t * params_st)
 {
-  void * ret_val;
-  data_types_t val_type;
-  int res = RCL_RET_OK;
-  rcl_allocator_t allocator;
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(seq_data_type, RCUTILS_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(params_st, RCUTILS_RET_INVALID_ARGUMENT);
+
+  rcutils_allocator_t allocator = params_st->allocator;
+  RCUTILS_CHECK_ALLOCATOR_WITH_MSG(
+    &allocator, "invalid allocator", return RCUTILS_RET_INVALID_ARGUMENT);
 
-  if ((NULL == params_st) || (0U == params_st->num_nodes) || (NULL == seq_data_type)) {
-    return RCL_RET_INVALID_ARGUMENT;
+  if (0U == params_st->num_nodes) {
+    RCUTILS_SET_ERROR_MSG("No node to update");
+    return RCUTILS_RET_INVALID_ARGUMENT;
   }
-  allocator = params_st->allocator;
 
   const size_t node_idx = (params_st->num_nodes - 1U);
   if (0U == params_st->params[node_idx].num_params) {
-    return RCL_RET_INVALID_ARGUMENT;
+    RCUTILS_SET_ERROR_MSG("No parameter to update");
+    return RCUTILS_RET_INVALID_ARGUMENT;
   }
 
   const size_t parameter_idx = ((params_st->params[node_idx].num_params) - 1U);
   const size_t val_size = event.data.scalar.length;
   const char * value = (char *)event.data.scalar.value;
   yaml_scalar_style_t style = event.data.scalar.style;
   const uint32_t line_num = ((uint32_t)(event.start_mark.line) + 1U);
-  rcl_variant_t * param_value;
+
+  if (NULL == value) {
+    RCUTILS_SET_ERROR_MSG("event argument has no value");
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
```

Comment:
and here

Reply:
Here it makes sense. Done in cf8bff1e0420edf110631ff124963d6a199f638f.